### ISSUE TITLE
Index signature incorrectly includes `undefined` when computed property name and spread are both present

### ISSUE THREAD

**Issue by RyanCavanaugh** [microsoft/TypeScript#39278]
**TypeScript Version:**  4.0.0-beta

**Search Terms:** index signature spread undefined computed

**Code**

```ts
declare const m: { [k: string]: string };
const x: { [k: string]: string } = { ...m, ["a" + "b"]: "" };
```

**Expected behavior:** No error (3.9.3)

**Actual behavior:** Error
```
a.ts:2:7 - error TS2322: Type '{ [x: string]: string | undefined; }' is not assignable to type '{ [k: string]: string; }'.
  Index signatures are incompatible.
    Type 'string | undefined' is not assignable to type 'string'.
      Type 'undefined' is not assignable to type 'string'.

2 const x: { [k: string]: string } = { ...m, ["a" + "b"]: "" };
        ~
```

**Playground Link:** [linky](https://www.typescriptlang.org/play/?ts=4.0.0-dev.20200625&ssl=3&ssc=1&pln=1&pc=1#code/CYUwxgNghgTiAEYD2A7AzgF3gWwFzwG94BtAa30xgEsUBzAXQo2rvgF8BuAKGXSwA98RMkxYNRNWu3gBeQvAB0S7ABoSAIijr4AanjqARusb7tnLkA)

**Related Issues:** no

**Comment by tchetwin** [comment]
Caught this in our CI earlier, was just about to enter an Issue, thanks!

Links to [3.9.2](https://www.typescriptlang.org/v2/en/play?ts=3.9.2#code/MYewdgzgLgBAhgJQKagE4BMBcNlvQHmlQEswBzAGhiNLID4YBeGAbwFgAoGb+bAIjgAjYHwqceMQf3RIAZqM4BfANydOoSLDgA5OAFsk2GuSYw+I1R3Xhok4mTJJUuEBmwuMhKCXJVj9U3YuHgA6MMQUV3QqAG0dfSQAXX4yAAtiPiVlIA) and [4.0.0-beta](https://www.typescriptlang.org/v2/en/play?ts=4.0.0-beta#code/MYewdgzgLgBAhgJQKagE4BMBcNlvQHmlQEswBzAGhiNLID4YBeGAbwFgAoGb+bAIjgAjYHwqceMQf3RIAZqM4BfANydOoSLDgA5OAFsk2GuSYw+I1R3Xhok4mTJJUuEBmwuMhKCXJVj9U3YuHgA6MMQUV3QqAG0dfSQAXX4yAAtiPiVlIA).
